fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -218,76 +204,140 @@ func configureFleetspeak(tempPath string, fsFrontendPort, fsAdminPort int, mysql
 	if err != nil {
 		return fmt.Errorf("Unable to create communicator.txt: %v", err)
 	}
-	err = ioutil.WriteFile(path.Join(tempPath, "textservices", "frr.textproto"), []byte(proto.MarshalTextString(&clientServiceConf)), 0644)
+	err = ioutil.WriteFile(path.Join(tempPath, "textservices", "frr.textproto"), []byte(proto.MarshalTextString(&clientServiceConfig)), 0644)
 	if err != nil {
 		return fmt.Errorf("Unable to write frr.textproto file: %v", err)
 	}
+	return nil
+}
+
+func configureFleetspeakServer(tempPath string, fsFrontendPort, fsHTTPSListenPort, fsAdminPort int, serverConfigPath, serverServicesConfigPath string) error {
```

Comment:
func modifyFleetspeakServerConfig(configDir, .., newServerConfigPath, newServerServicesConfigPath)

To clarify what the function does and what it produces; configureFoo is just too generic in this case).